tests/conftest.py

```diff
@@ -550,8 +550,10 @@ def get_requests(self):
         """Get environ for each received request.
         """
         assert not self._running, "cannot get mock from running server"
+        # TODO: Change call[0][0] to call.args[0] after dropping Python 3.7
+        # support.
         return [
-            call.args[0] for call in self._server.mock.call_args_list
+            call[0][0] for call in self._server.mock.call_args_list
```

Comment:
The call class on 3.8+ has a three-element form, and the first element is `name`. So this needs to be `call[-2][0]` 😞

Reply:
But I wonder if it’d be cleaner if we implement something by hooking into `side_effect` to make `_MockServer` capture requests instead, and avoid mucking with mock internals.

Reply:
Interestingly the CI passes on 3.8 and 3.9, but not on 3.6 and 3.7.

Reply:
@uranusjr Thanks! I wasn't aware of that change. I'm using `[-2]` in the latest revision.

The 3.6 failure is due to another `call` object in `tests/functional/test_install.py`. This has been fixed in the latest revision.